kphp_polyfills.php

```diff
@@ -65,7 +65,8 @@ function not_null($any_value) {
   if ($any_value === null) {
     warning("Passed 'null' to not_null() in PHP");
   }
-  return $any_value;
+  // turn off PhpStorm native inferring
```

Comment:
For the record, PhpStorm 2020.3 added Psalm support, so instead of this hack you can use a proper PHPDoc:

```php
/**
 * @template T
 * @param T|null $any_value
 * @return T
 */
function not_null($any_value) {
```

See details [here](https://blog.jetbrains.com/phpstorm/2020/12/phpstorm-2020-3-release/#psalm_and_phpstan_support).

![example](https://blog.jetbrains.com/wp-content/uploads/2020/10/psalm_types_completion.png)

Reply:
Thanks for the advice, some time we'll look at simplifying our PhpStorm plugin so that more PhpStorm features are used natively